tools/clang/include/clang/SPIRV/SpirvType.h

```diff
@@ -139,9 +137,15 @@ class MatrixType : public SpirvType {
   const SpirvType *getVecType() const {
     return llvm::cast<SpirvType>(vectorType);
   }
+  const SpirvType *getElementType() const {
+    return vectorType->getElementType();
+  }
   uint32_t getVecCount() const { return vectorCount; }
   bool isRowMajorMat() const { return isRowMajor; }
 
+  uint32_t numRows() const { return vectorCount; }
```

Comment:
Hmm, we should be a little bit cautious here. The majorness difference between HLSL and SPIR-V is quite tricky. Conceputally, SPIR-V is column-major, which means `vectorCount` is `numCols` here.